### PR TITLE
Edit project ( post and deleted images)

### PR DESCRIPTION
**Pull request by Melody-Le**

### CODE REVIEW

File: controllers/projectController.js [hunk 1]
```diff
@@ -102,7 +115,44 @@ const controller = {
...
 
   editProject: async (req, res) => {
+    // check If user change project images and logo?
+    let newProjectImages = [];
+    const deletedProjectImages = req.body.deletedImages;
+    if (req.files) {
+      try {
+        req.body.logo_url = await getLogoUrl(req.files, req.body.slug);
+      } catch (error) {
+        console.log(error);
```
Comment (ZoeNguyen73) [inline controllers/projectController.js]: can remove this console.log

Comment (ZoeNguyen73) [inline controllers/projectController.js]: can remove console.log here as well!

File: controllers/projectController.js [hunk 2]
```diff
@@ -102,7 +115,44 @@ const controller = {
...
+    const oldProjectData = await ProjectModel.findOne(
+      { slug: req.params.slug },
+      { image_urls: 1, _id: 0 }
+    );
+    const oldImages = oldProjectData?.image_urls;
+    //if user dont delete any image:
+    let oldImagesAfterDelete = oldImages;
+    //if user delete some images:
+    if (deletedProjectImages) {
+      oldImagesAfterDelete = diffArray(oldImages, deletedProjectImages);
```
Comment (ZoeNguyen73) [inline controllers/projectController.js:147]: diffArr = [...arr1, ...arr2].filter((v) => arr1.includes(v) !== arr2.includes(v));

if
arr1 = [a,b]
arr2 = [b,c]
diffArr will be [a,c] ... which is not expected here right? as we only want [a]
should we change the diffArr logic to filter((v) => arr1.includes(v) && !arr2.includes(v)) ?

Comment (Melody-Le) [reply controllers/projectController.js:147]: Updated

File: controllers/projectController.js (hunk 1 above)
Comment (Melody-Le) [reply controllers/projectController.js]: updated

**Review by ZoeNguyen73 — APPROVED** [review]

**Pull request merged**